packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -122,6 +123,14 @@ const TreeItem = React.forwardRef(function TreeItem(props, ref) {
     treeId,
   } = React.useContext(TreeViewContext);
 
+  let id = null;
+
+  if (idProp) {
```

Comment:
For my own piece of mind: Can we remove the weak casting to boolean here? It's just confusing long term. 
```suggestion
  if (idProp != null) {
```

Same for the else-if branch